src/cluster_legacy.c

```diff
@@ -2791,6 +2798,8 @@ int clusterProcessPacket(clusterLink *link) {
 
     sender = getNodeFromLinkAndMsg(link, hdr);
 
+    if (sender) sender->extensions_supported = (hdr->mflags[0] & CLUSTERMSG_FLAG0_NODE_SUPPORTS_EXTENSIONS);
```

Comment:
existing 7.2/7.0 will not carry `CLUSTERMSG_FLAG0_NODE_SUPPORTS_EXTENSIONS` hence they would be marked as "not capable of supporting extensions", which is not accurate. Shouldn't we check `hdr->mflags[0] & CLUSTERMSG_FLAG0_EXT_DATA` instead?